File: src/my_project_typescript/easy_problems_typescript/from1to50/replaceAllDigitsWithCharacters.ts
function replaceDigits(s: string): string {

    let temp: string = ""
    let answer: string = ""
    let x: number = 0
    
    for (let i = 1; i < s.length; i = i + 2){
        x = s[i-1].charCodeAt(0)
        temp = String.fromCharCode(x + parseInt(s[i]))
        answer += s[i-1] + temp 
    }

    if (s.length % 2 === 1){
        answer += s[s.length-1]
    }

    return answer

};


console.log(replaceDigits("a1c1e1"))

console.log(replaceDigits("a1b2c3d4e"))

File: src/my_project/easy_problems/from101to150/replace_all_digits_with_characters.py
from typing import List 

class Solution:
    def replaceDigits(self, s: str) -> str:

        temp: str = ""
        answer: str = ""

        for character in range(1, len(s), 2):
            x = ord(s[character-1])
            temp = chr(x + int(s[character]))
            answer += s[character-1] + temp 

        if len(s)%2 ==1:
            answer += s[len(s)-1]


        return answer 


solution = Solution()

print(solution.replaceDigits(s = "a1c1e1"))

print(solution.replaceDigits( s = "a1b2c3d4e"))



